synapseutils/describe.py

```diff
@@ -0,0 +1,58 @@
+import synapseclient
+import pandas as pd
+
+
+def _open_entity_as_df(syn, entity: str) -> pd.DataFrame:
+    """
+    :param syn: synapse object
+    :param entity: a synapse entity to be extracted and converted into a dataframe
+    :return: a pandas DataFrame if flow of execution is successful; None if not.
+    """
+
+    dataset = None
+
+    try:
+        entity = syn.get(entity)
+        format = entity.path.split(".")[-1]
+    except synapseclient.core.exceptions.SynapseHTTPError:
+        print(str(entity) + " is not a valid Synapse id")
+        return dataset  # its value is None here
+
+    if format == "csv":
+        dataset = pd.read_csv(entity.path)
+    elif format == "tsv":
+        dataset = pd.read_csv(entity.path, sep='\\t')
+    else:
+        print("File type not supported.")
+
+    return dataset
+
+
+def synapse_describe(df: pd.DataFrame, mode: str = 'string'):
+    """
+    :param df: pandas dataframe from the csv or tsv file
+    :param mode: string defining the return value.  Can be either 'dict' or 'string'
+    :return: see param mode
+    """
+
+    stats = {}
+    for column in df.columns:
+        stats[column] = {}
```

Comment:
Not a super important change either but you can use `collections.defaultdict` here
```suggestion
    stats = defaultdict(dict)
    for column in df.columns:
```

Reply:
I like this better too.  I changed it.